WooCommerce/Classes/ViewRelated/Orders/Order Creation/NewOrder.swift

```diff
@@ -8,14 +8,44 @@ final class NewOrderHostingController: UIHostingController<NewOrder> {
     /// References to keep the Combine subscriptions alive within the lifecycle of the object.
     ///
     private var subscriptions: Set<AnyCancellable> = []
+    private let viewModel: NewOrderViewModel
 
     init(viewModel: NewOrderViewModel) {
+        self.viewModel = viewModel
         super.init(rootView: NewOrder(viewModel: viewModel))
     }
 
     required dynamic init?(coder aDecoder: NSCoder) {
         fatalError("init(coder:) has not been implemented")
     }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        if ServiceLocator.featureFlagService.isFeatureFlagEnabled(.splitViewInOrdersTab) {
+            // Set presentation delegate to track the user dismiss flow event
+            if let navigationController = navigationController {
+                navigationController.presentationController?.delegate = self
+            } else {
+                presentationController?.delegate = self
+            }
+        }
+    }
+}
+
+/// Intercepts to the dismiss drag gesture.
+///
+extension NewOrderHostingController: UIAdaptivePresentationControllerDelegate {
+    func presentationControllerShouldDismiss(_ presentationController: UIPresentationController) -> Bool {
+        guard viewModel.hasChanges else {
+            return true
+        }
+        UIAlertController.presentDiscardChangesActionSheet(viewController: self, onDiscard: { [weak self] in
+            self?.dismiss(animated: true, completion: nil)
+        })
```

Comment:
For a slightly better experience, this bit can go in the `presentationControllerDidAttemptToDismiss` which is called after this method return `false` and gives a nice bounce feedback to the user.

Reply:
Good idea! Updated in 2ba3f08901899bc1d77aa0e8181273c3ad564cdc.